Magnifier implementation (#19228)
nvaccess/nvda: user_docs/en/userGuide.md

It provides several configuration options to customize the magnification experience according to your needs.

### Enabling and Disabling the Magnifier {#MagnifierToggle}

To enable or disable the magnifier, press `NVDA+shift+w`.
When the magnifier is enabled, NVDA will announce the current zoom level, color filter, and focus tracking mode.
When disabled, the screen returns to its normal size.

Important: The NVDA Magnifier cannot be used simultaneously with Screen Curtain for security reasons.
If you attempt to enable the magnifier while Screen Curtain is active, NVDA will prompt you to disable Screen Curtain first, and vice versa.

### Magnifier Controls {#MagnifierControls}

Once the magnifier is enabled, you can use the following keyboard commands to control it:

<!-- KC:beginInclude -->

| Name |Key |Description|
|---|---|---|
|Toggles the magnifier on and off |`NVDA+shift+w` |Enables or disables the magnifier|
|Increases the magnification level of the magnifier |`NVDA+shift+equals` |Increases the zoom level|
|Decreases the magnification level of the magnifier |`NVDA+shift+minus` |Decreases the zoom level|

> |Toggle filter of the magnifier | None |Cycles through available color filters (normal, grayscale, inverted)|

Reviewer: @Boumtchack - can you do this in a separate PR? I think `nvda+shift+i` is fine

### Enabling and Disabling the Magnifier {#MagnifierToggle}

To enable or disable the magnifier, press `NVDA+shift+w`.
When the magnifier is enabled, NVDA will announce the current zoom level, color filter, and focus tracking mode.
When disabled, the screen returns to its normal size.

Important: The NVDA Magnifier cannot be used simultaneously with Screen Curtain for security reasons.
If you attempt to enable the magnifier while Screen Curtain is active, NVDA will prompt you to disable Screen Curtain first, and vice versa.

### Magnifier Controls {#MagnifierControls}

Once the magnifier is enabled, you can use the following keyboard commands to control it:

<!-- KC:beginInclude -->

| Name |Key |Description|
|---|---|---|
|Toggle magnifier |`NVDA+shift+w` |Enables or disables the magnifier|
|Zoom in |`NVDA+shift+upArrow` |Increases the zoom level|
|Zoom out |`NVDA+shift+downArrow` |Decreases the zoom level|
|Toggle color filter |`NVDA+shift+f` |Cycles through available color filters (normal, grayscale, inverted)|
|Toggle focus mode |`NVDA+shift+m` |Cycles through focus tracking modes (center, border, relative)|

> |Start spotlight mode |`NVDA+shift+s` |Activates spotlight mode for focused reading or presentations|

Reviewer: ```suggestion
|Toggles the magnifier on and off |`NVDA+shift+w` |Enables or disables the magnifier|
|Increases the magnification level of the magnifier |`NVDA+shift+=` |Increases the zoom level|
|Decreases the magnification level of the magnifier |`NVDA+shift+-` |Decreases the zoom level|
|Toggle filter of the magnifier | None |Cycles through available color filters (normal, grayscale, inverted)|
|Toggle focus mode for the full-screen magnifier | None |Cycles through focus tracking modes (center, border, relative)|
|Launch spotlight if magnifier is full-screen |`NVDA+shift+l` |Activates spotlight mode for focused reading or presentations|
```

<!-- KC:endInclude -->

### Zoom Levels {#MagnifierZoomLevels}

> The magnifier supports zoom levels from 1.0 (no magnification) to 10.0 (maximum magnification).

Reviewer 2: This line gives the maximum magnification as 10.0.
Line 2792 gives the maximum magnificatio as 16.0

You can adjust the zoom level using the zoom in (`NVDA+shift+upArrow`) and zoom out (`NVDA+shift+downArrow`) commands.
Each press increases or decreases the zoom by a fixed increment.

The default zoom level when the magnifier is first enabled can be configured in the [Magnifier settings](#MagnifierSettings).

### Color Filters {#MagnifierColorFilters}

Color filters can help users with certain visual impairments or light sensitivity by modifying the colors displayed on the screen.
The magnifier provides three color filter options:

* **Normal**: No color modification is applied. This is the default setting.
* **Grayscale**: Converts all colors to shades of gray, which can help reduce eye strain and improve contrast for some users.
* **Inverted**: Inverts all colors on the screen (black becomes white, white becomes black, etc.), which can be helpful for users who prefer light text on dark backgrounds or have photophobia.

> To cycle through the available filters, press `NVDA+shift+f`.

Reviewer: ```suggestion
To cycle through the available filters, please assign a custom gesture using the [Input Gestures dialog](#InputGestures).
```

This mode keeps the focused element at the center of the screen at all times.
* **Border**: The magnified area only moves when the focus approaches the edge of the visible area.
This mode provides a more stable view, only adjusting when necessary.
* **Relative**: The magnified area maintains the relative position of the focus within the screen.
This mode mimics the behavior of the Windows Magnifier.

To cycle through the focus tracking modes, press `NVDA+shift+m`.
NVDA will announce the name of the currently selected mode.

The default focus mode when the magnifier is first enabled can be configured in the [Magnifier settings](#MagnifierSettings).

### Spotlight Mode {#MagnifierSpotlight}

Spotlight mode is a special feature designed for presentations or focused reading tasks.
When activated, it temporarily zooms out the magnified view to show the full screen, then zooms back in to the current focus position after a brief period of mouse inactivity.

This is useful when you want to:

* Show context to your audience during a presentation before zooming in on specific details
* Temporarily view the full screen layout while magnifying

> To activate spotlight mode, press `NVDA+shift+s` while the magnifier is enabled.

Reviewer: ```suggestion
To activate spotlight mode, press `NVDA+shift+l` while the magnifier is enabled.
```

The default value is 2.0 (200% zoom).

You can always adjust the zoom level on the fly using the zoom in (`NVDA+shift+upArrow`) and zoom out (`NVDA+shift+downArrow`) commands while the magnifier is active.

##### Default color filter {#MagnifierDefaultFilter}

This combo box allows you to select the default color filter to apply when the magnifier is first enabled.
The available options are:

* Normal: No color modification is applied. This is the default setting.
* Grayscale: Converts all colors to shades of gray, which can help reduce eye strain and improve contrast.
* Inverted: Inverts all colors on the screen, which can be helpful for users who prefer light text on dark backgrounds or have photophobia.

You can cycle through the color filters on the fly using `NVDA+shift+f` while the magnifier is active.

##### Default focus mode {#MagnifierDefaultFocusMode}

This combo box allows you to select the default focus tracking mode when the magnifier is first enabled.
The available options are:

* Center: The magnified area is always centered on the current focus position.
* Border: The magnified area only moves when the focus approaches the edge of the visible area.

> * Relative: The magnified area maintains the relative position of the focus within the screen.

Reviewer: same here retables

When these providers have adjustable settings, they will be shown in this settings category in separate groupings.
For the supported settings per provider, please refer to the documentation for that provider.

#### Magnifier {#MagnifierSettingsCategory}

<!-- KC:setting -->

##### Open Magnifier settings {#OpenMagnifierSettings}

> Key: `NVDA+control+shift+m`

Reviewer: ```suggestion
Key: `NVDA+control+w`
```

The Magnifier category in the NVDA Settings dialog allows you to configure the default behavior of NVDA's built-in [Magnifier](#Magnifier) feature.
This settings category contains the following options:

##### Default zoom level {#MagnifierDefaultZoom}

This slider allows you to set the default zoom level when the magnifier is first enabled.
The zoom level can range from 1.0 (no magnification) to 16.0 (maximum magnification).
The default value is 2.0 (200% zoom).

You can always adjust the zoom level on the fly using the zoom in (`NVDA+shift+upArrow`) and zoom out (`NVDA+shift+downArrow`) commands while the magnifier is active.

| . {.hideHeaderRow} |.|
|---|---|
|Options |1.0 to 16.0|
|Default |2.0|

##### Default color filter {#MagnifierDefaultFilter}

This combo box allows you to select the default color filter to apply when the magnifier is first enabled.
You can cycle through the color filters on the fly using `NVDA+shift+f` while the magnifier is active.
The available options are:

| . {.hideHeaderRow} |.|
|---|---|
|Options | Normal, Grayscale, Inverted |
|Default |Normal |

> |Toggle command |`NVDA+shift+f` |

Reviewer: ```suggestion
```

| Option | Description |
|---|---|
| Normal | No color modification is applied. |
| Grayscale | Converts all colors to shades of gray, which can help reduce eye strain and improve contrast. |
| Inverted | Inverts all colors on the screen, which can be helpful for users who prefer light text on dark backgrounds or have photophobia. |

##### Default focus mode {#MagnifierDefaultFocusMode}

This combo box allows you to select the default focus tracking mode when the magnifier is first enabled.
The available options are:

| . {.hideHeaderRow} |.|
|---|---|
|Options |Center, Border, Relative|
|Default |Center|

> |Toggle command |`NVDA+shift+m`|

Reviewer: ```suggestion
```

| Option | Description |
|---|---|
| Center | The magnified area is always centered on the current focus position. |
| Border | The magnified area only moves when the focus approaches the edge of the visible area. |
| Relative | The magnified area maintains the relative position of the focus within the screen. |

You can cycle through the focus modes on the fly using `NVDA+shift+m` while the magnifier is active.

> 

Reviewer: ```suggestion
```

This combo box allows you to select the default color filter to apply when the magnifier is first enabled.
The available options are:

> | Option | Description |

Reviewer: please also add the table that summaries the options defaults e.g. as per our [user guide standards](https://github.com/nvaccess/nvda/blob/master/projectDocs/dev/userGuideStandards.md#feature-settings). See [BrailleFormattingDisplay](https://download.nvaccess.org/releases/beta/documentation/userGuide.html#BrailleFormattingDisplay) in the user guide for an example

```
| . {.hideHeaderRow} |.|
|---|---|
|Options |Default (Enabled), Disabled, Enabled |
|Default |Enabled |
|Toggle command |`NVDA+shift+e` |
```

##### Default color filter {#MagnifierDefaultFilter}

> | . {.hideHeaderRow} |.|

Reviewer: please move this to below the summary paragraph, but above the other table

|---|---|
|Options |Default (Enabled), Disabled, Enabled |

> |Default |Enabled |

Reviewer: please make sure to list the actual options and defaults

|Toggle command |`NVDA+shift+e` |

This combo box allows you to select the default color filter to apply when the magnifier is first enabled.
The available options are:

| Option | Description |
|---|---|
| Normal | No color modification is applied. |
| Grayscale | Converts all colors to shades of gray, which can help reduce eye strain and improve contrast. |
| Inverted | Inverts all colors on the screen, which can be helpful for users who prefer light text on dark backgrounds or have photophobia. |

You can cycle through the color filters on the fly using `NVDA+shift+f` while the magnifier is active.

##### Default focus mode {#MagnifierDefaultFocusMode}

> | . {.hideHeaderRow} |.|

Reviewer: please move this to below the summary paragraph, but above the other table

##### Keep mouse centered {#MagnifierKeepMouseCentered}

| . {.hideHeaderRow} |.|
|---|---|

> |Options |Default (Disabled), Enabled, Disabled|

Reviewer: ```suggestion
|Options |Disabled, Enabled|
```
